src/assessments/seo/introductionKeywordAssessment.js

```diff
@@ -1,62 +1,99 @@
-var AssessmentResult = require( "../../values/AssessmentResult.js" );
+const Assessment = require( "../../assessment.js" );
+const AssessmentResult = require( "../../values/AssessmentResult.js" );
+const merge = require( "lodash/merge" );
 
 /**
- * Returns a score and text based on the firstParagraph object.
- *
- * @param {object} firstParagraphMatches The object with all firstParagraphMatches.
- * @param {object} i18n The object used for translations
- * @returns {object} resultObject with score and text
+ * Assessment to check whether the keyphrase is encountered in the first paragraph of the article.
  */
-var calculateFirstParagraphResult = function( firstParagraphMatches, i18n ) {
-	const url = "<a href='https://yoa.st/2pc' target='_blank'>";
+class IntroductionHasKeywordAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		let defaultConfig = {
+			parameters: {
+				recommendedMinimum: 1,
+			},
+			scores: {
+				good: 9,
+				bad: 3,
+			},
+			url: "<a href='https://yoa.st/2pc' target='_blank'>",
+		};
+
+		this.identifier = "introductionKeyword";
+		this._config = merge( defaultConfig, config );
+	}
+
+	/**
+	 * Assesses the presence of keyphrase in the first paragraph.
+	 *
+	 * @param {Paper} paper The paper to use for the assessment.
+	 * @param {Researcher} researcher The researcher used for calling research.
+	 * @param {Jed} i18n The object used for translations.
+	 *
+	 * @returns {AssessmentResult} The result of this assessment.
+	 */
+	getResult( paper, researcher, i18n ) {
+		let assessmentResult = new AssessmentResult();
+
+		this._firstParagraphMatches = researcher.getResearch( "firstParagraph" );
```

Comment:
* About `getResarch`: true, it would more sense if this were called `getResearchResults`. However, I think this falls outside the scope of the refactoring in this PR.
* About `calculatereResult`: this is saved as a class variable in order to keep the `calculateResult` methods consistent across assessments.